modules/firefox/each-config.nix

```
@@ -0,0 +1,154 @@
+# manually applied args
+{
+  mkTarget,
+  name,
+  humanName,
+}:
+# module args
+{
+  lib,
+  pkgs,
+  ...
+}:
+mkTarget {
+  inherit name humanName;
+
+  extraOptions = {
+    profileNames = lib.mkOption {
+      description = "The ${humanName} profile names to apply styling on.";
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+    };
+
+    colorTheme.enable = lib.mkEnableOption "[Firefox Color](https://color.firefox.com/) on ${humanName}";
+
+    firefoxGnomeTheme.enable = lib.mkEnableOption "[Firefox GNOME theme](https://github.com/rafaelmardojai/firefox-gnome-theme) on ${humanName}";
+  };
+
+  configElements = [
+    (
+      { cfg }:
+      {
+        warnings =
+          lib.optional (cfg.profileNames == [ ])
+            ''stylix: ${name}: `config.stylix.targets.${name}.profileNames` is not set. Declare profile names with 'config.stylix.targets.${name}.profileNames = [ "<PROFILE_NAME>" ];'.'';
+      }
+    )
+    (
+      { cfg, fonts }:
+      {
+        programs.${name}.profiles = lib.genAttrs cfg.profileNames {
+          settings = {
+            "font.name.monospace.x-western" = fonts.monospace.name;
+            "font.name.sans-serif.x-western" = fonts.sansSerif.name;
+            "font.name.serif.x-western" = fonts.serif.name;
+          };
+        };
+      }
+    )
+    (
+      {
+        cfg,
+        colors,
+        inputs,
+      }:
+      {
+        programs.${name}.profiles = lib.genAttrs cfg.profileNames (
+          lib.mkIf cfg.firefoxGnomeTheme.enable {
+            settings = {
+              "toolkit.legacyUserProfileCustomizations.stylesheets" = true;
+              "svg.context-properties.content.enabled" = true;
+            };
+
+            userChrome =
+              let
+                template = colors {
+                  template = ./userChrome.css.mustache;
+                  extension = ".css";
+                };
+              in
+              ''
+                @import "${inputs.firefox-gnome-theme}/userChrome.css";
+                @import "${template}";
+              '';
+
+            userContent = ''
+              @import "${inputs.firefox-gnome-theme}/userContent.css";
+            '';
+          }
+        );
+      }
+    )
+    (
+      {
+        cfg,
+        colors,
+        inputs,
+      }:
+      let
+        mkColor = color: {
+          r = colors."${color}-rgb-r";
+          g = colors."${color}-rgb-g";
+          b = colors."${color}-rgb-b";
+        };
+        inherit
+          (inputs.nur.legacyPackages.${pkgs.stdenv.hostPlatform.system}.repos.rycee)
+          firefox-addons
+          ;
+      in
+      {
+        programs.${name}.profiles = lib.genAttrs cfg.profileNames (
+          lib.mkIf cfg.colorTheme.enable {
```

1. pull the `mkIf` up a level
2. genAttrs takes a function

```suggestion
        programs.${name}.profiles = lib.mkIf cfg.colorTheme.enable (lib.genAttrs cfg.profilesNames (_:
           {
```

Or switch to `mkIf cfg.colorTheme.enable (mkMerge (map … cfg.profilesNames))`